*** dlwrap/posix.c
--- dlwrap/posix.c
***************
*** 22,28 ****
  
  #include "os.h"
  
! #include "nscq/nscq.h"
  
  #include <dlfcn.h>
  #include <errno.h>
--- 22,28 ----
  
  #include "os.h"
  
! #include "nscq.h"
  
  #include <dlfcn.h>
  #include <errno.h>
***************
*** 55,73 ****
      int cycle;
  } nscq_dl = {PTHREAD_RWLOCK_INITIALIZER, NULL, 0};
  
! // Loads the requested major version of the library using dlopen().
! int nscq_dl_load() {
!     char dlname[32];
      int ret;
  
-     ret = snprintf(dlname, sizeof(dlname), "libnvidia-nscq.so.%u",
-                    NSCQ_API_VERSION_CODE_MAJOR(NSCQ_API_VERSION_CODE));
-     if (ret < 0) {
-         return ret;
-     } else if (((size_t)ret) >= sizeof(dlname)) {
-         return -ENAMETOOLONG;
-     }
- 
      ret = pthread_rwlock_wrlock(&nscq_dl.rwlock);
      if (ret != 0) {
          return -ret;
--- 55,64 ----
      int cycle;
  } nscq_dl = {PTHREAD_RWLOCK_INITIALIZER, NULL, 0};
  
! // Helper for nscq_dl_load()
! int nscq_dl_load_helper(char *dlname) {
      int ret;
  
      ret = pthread_rwlock_wrlock(&nscq_dl.rwlock);
      if (ret != 0) {
          return -ret;
***************
*** 86,91 ****
--- 77,96 ----
  
      return ret;
  }
+ 
+ // Loads the requested major version of the library using dlopen() or the
+ // DCGM-provided library if standalone is not found on the system.
+ int nscq_dl_load() {
+     int ret;
+ 
+     ret = nscq_dl_load_helper("libnvidia-nscq.so");
+ 
+     if (ret == -ELIBACC) {
+         ret = nscq_dl_load_helper("libnvidia-nscq-dcgm.so");
+     }
+ 
+     return ret;
+ }
  
  // Unloads the NSCQ library loaded by nscq_dl_load(), using dlclose().
  void nscq_dl_unload(void) {
*** dlwrap/dlwrap.c
--- dlwrap/dlwrap.c
***************
*** 20,26 ****
  // DEALINGS IN THE SOFTWARE.
  //

! #include "nscq/dlwrap.h"

  #include "os.h"

--- 20,26 ----
  // DEALINGS IN THE SOFTWARE.
  //

! #include "dlwrap.h"

  #include "os.h"

*** dlwrap/dlwrap.h
--- dlwrap/dlwrap.h
***************
*** 26,32 ****
  extern "C" {
  #endif

! #include "nscq/nscq.h"

  int nscq_dlwrap_attach(void);
  void nscq_dlwrap_detach(void);
--- 26,32 ----
  extern "C" {
  #endif

! #include "nscq.h"

  int nscq_dlwrap_attach(void);
  void nscq_dlwrap_detach(void);
